src/clib/pio_file.c

```diff
@@ -452,9 +451,9 @@ int PIOc_deletefile(int iosysid, const char *filename)
 
     /* Broadcast and check the return code. */
     if ((mpierr = MPI_Bcast(&ierr, 1, MPI_INT, ios->ioroot, ios->my_comm)))
-        return check_mpi(file, mpierr2, __FILE__, __LINE__);        
+        return check_mpi(NULL, mpierr2, __FILE__, __LINE__);        
     if (ierr)
-        return check_netcdf(file, ierr, __FILE__, __LINE__);
+        return check_netcdf(NULL, ierr, __FILE__, __LINE__);
```

Comment:
A new version of check_netcdf() is needed. I checked out develop branch and merged this PR for testing, pio_unit_test still failed due to an assertion failure.
pio_unit_test: src/clib/pioc_support.c:366: check_netcdf: Assertion `file' failed.